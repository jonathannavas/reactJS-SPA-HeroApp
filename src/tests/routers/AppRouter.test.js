import { mount } from "enzyme";
import { AuthContext } from "../../auth/AuthContext";
import { AppRouter } from "../../routers/AppRouter";

describe('Tests on AppRouter.js', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    test('Should show the login page if is not authenticated', () => {

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                 <AppRouter />
            </AuthContext.Provider>     
        );

        expect( wrapper ).toMatchSnapshot();

    });

    test('Should to show marvel componente if the user is authenticated', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Jonathan'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                 <AppRouter />
            </AuthContext.Provider>     
        );
        
        expect( wrapper.find('.navbar').exists() ).toBe(true);
    });

});