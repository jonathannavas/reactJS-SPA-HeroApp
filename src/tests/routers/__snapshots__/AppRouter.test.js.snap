// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on AppRouter.js Should show the login page if is not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Ingresar
    </button>
  </div>
</div>
`;
